Apex/Assignment 2/src/classes/TrainingSessionManager.cls

```diff
@@ -0,0 +1,287 @@
+    }
+    /**
+     * @description: Use to Fetch Attendee__c count on a  Training Session.
+     */
+    public void FetchRecordsBasedOnCount() {
+        AggregateResult attendeeCount = [SELECT Count(Id) totalCount FROM Attendee__c where Training_Session__c = 'a0GdM0000084MMPUA2'];
+        System.debug(attendeeCount.get('totalCount'));
+    }
+    /**
+     * @description: Use to Fetch Records based on Record Type.
+     */
+    public void FetchRecordsBasedOnRecordType() {
+        //Getting Webinar RT ID 
+        String gsRecordTypeID = Schema.SObjectType.Training_Session__c.getRecordTypeInfosByName().get('Webinar RT').getRecordTypeId();
+		System.debug(gsRecordTypeID);
+        List<Training_Session__c> recordBasedSessions = [SELECT Id, Name FROM Training_Session__c where RecordTypeId = :gsRecordTypeID];
+    }
+    
+	/**
+     * @description: Creates a list of dummy Training_Session__c records for bulk operations.
+     * @return: A List<Training_Session__c> containing dummy training session records.
+     */
+	public List<Training_Session__c> createList(){
+   		List<Training_Session__c> ListToInsertUsingBulkOperation = new List<Training_Session__c>();
+        for(Integer i=1;i<10;i++) {
+          Training_Session__c Student = new Training_Session__c();
+          Student.Name = 'FEXLE' + i;
+          ListToInsertUsingBulkOperation.add(Student);	
+    	}
+        
+        return ListToInsertUsingBulkOperation;
+	}
+    
+    /**
+     * @description: Deletes a set of Training_Session__c records in bulk.
+     */
+    public void deleteBulkRecords() {
+        List<Training_Session__c> sessions = [SELECT ID FROM Training_Session__c LIMIT 5];
+        if (!sessions.isEmpty()) {
+            Database.DeleteResult[] DR_Dels = Database.delete(sessions, false);
+            for (Database.DeleteResult result : DR_Dels) {
+                if (result.isSuccess()) {
+                    System.debug('Status: ' + Status.Success + ', Record ID: ' + result.getId());
+                } else {
+                    for (Database.Error error : result.getErrors()) {
+                        System.debug('Status: ' + Status.Failed + ', Error: ' + error.getMessage());
+                    }
+                }
+            
+         	}
+        }
+    }
+    /**
+     * @description: Inserts a list of Training_Session__c records in bulk.
+     */
+    public void insertBulkRecords() {
+        List<Training_Session__c> sessions = createList();
+        List<Database.SaveResult> insertSessions = Database.insert(sessions, false);
+        for (Database.SaveResult result : insertSessions) {
+            if (result.isSuccess()) {
+                System.debug('Status: ' + Status.Success + ', Record ID: ' + result.getId());
+            } else {
+                for (Database.Error error : result.getErrors()) {
+                    System.debug('Status: ' + Status.Failed + ', Error: ' + error.getMessage());
```

Comment:
Suppose we get errors at 5 instances, then it will show 5 debugs. Now try to find a way on which I can get the details of all 5 failures in a single debug.